src/Symfony/Component/HttpFoundation/ParameterBag.php

```diff
@@ -188,6 +188,35 @@ public function getBoolean($key, $default = false)
     }
 
     /**
+     * Returns the parameter value converted to a DateTime object.
+     *
+     * @param string             $key      The parameter key
+     * @param string             $format   The expected date format
+     * @param string             $default  The default value to be converted to a DateTime object if the parameter key does not exist
+     * @param \DateTimeZone|null $timeZone A DateTimeZone object representing the desired time zone
+     *
+     * @return \DateTime|null
+     */
+    public function getDate($key, $format, $default = null, $timeZone = null)
+    {
+        $time = $this->get($key, (string) $default);
```

Comment:
Not too sold on creating a variable and checking for value in the same line, however that is a matter of preference. 

Early return might be doable . However I avoided it because fabbot doesn't accept `return null;` but wants just a `return;` which technically both return `null` but I still see it as a `void` vs `null` issue. 

Regarding `\LogicException`: None of the methods of this class throws an exception for invalid input. (e.g.: `getInt`  rather than checking if it's a valid number or not it simply type-casts it).